src/Core/Silk.NET.Core/Native/ComPtr.cs

```diff
@@ -43,13 +43,17 @@ public uint Release()
             return ret;
         }
 
-        public readonly ref T* GetPinnableReference()
-        {
-            fixed (T** ptr = &Handle)
-            {
-                return ref *ptr;
-            }
-        }
+        // 1. get a ref for this, which is a readonly ref when within a readonly function
+        // 2. interpret that ref as a reference to a T. it's actually a ref to a T* (well actually it's a ComPtr<T>),
+        //    but let's roll with it as it's as good as we're gonna get without having pointers as generic type
+        //    parameters
+        // 3. get a T* from that "reference to a T".
+        // 4. cast it to a T**, which is its true identity.
+        public readonly T** GetAddressOf()
+            => (T**) Unsafe.AsPointer(ref Unsafe.As<ComPtr<T>, T>(ref Unsafe.AsRef(in this)));
```

Comment:
isn't the middle Unsafe.As redundant? I believe that's just erased by the pointer cast?

Reply:
Good point
```suggestion
            => (T**) Unsafe.AsPointer(ref Unsafe.AsRef(in this));
```